pre_commit/repository.py

```diff
@@ -64,11 +64,10 @@ def languages(self):
 
     @cached_property
     def additional_dependencies(self):
-        dep_dict = defaultdict(lambda: defaultdict(set))
+        dep_dict = defaultdict(lambda: defaultdict(list))
         for _, hook in self.hooks:
-            dep_dict[hook['language']][hook['language_version']].update(
-                hook.get('additional_dependencies', []),
-            )
+            dep_dict[hook['language']][hook['language_version']] += \
+                hook.get('additional_dependencies', [])
```

Comment:
The form is fine, I don't want to add language-specific parsing of this stuff if avoidable.

Heh, if we could only target python3.6 (or modern pypy), the set would have insertion order